unity-renderer/Assets/DCLPlugins/ECS7/ECSComponents/AvatarShape/Handler/AvatarShapeComponentHandler.cs

```diff
@@ -46,7 +60,13 @@ public void OnComponentModelUpdated(IParcelScene scene, IDCLEntity entity, PBAva
                 avatar.Init();
                 isAvatarInitialized = true;
             }
-            avatar.ApplyModel(scene,entity,model);
+            avatar.ApplyModel(scene, entity, model);
+        }
+
+        private void OnCombinedRendererUpdate(Renderer newRenderer)
+        {
+            renderersInternalComponent.RemoveFor(scene, entity);
+            renderersInternalComponent.AddRenderer(scene, entity, newRenderer);
```

Comment:
why are removing the component for that entity and on the next line setting the renderers for the component?